### PR TITLE
fix vive crash in AMD driver

### PR DESCRIPTION
**Pull request by zzmp**

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 5ab5cf1fe9eb3b5f9afd75370cba4f9a63ba5069 available at: https://gist.github.com/ad709337264fa1fc6dd3302f9c007a8a<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8476-5ab5cf1.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8476-5ab5cf1.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: ec63258f8556fbf5fbb93769dbeb0cf90b81b5a6 available at: https://gist.github.com/ad709337264fa1fc6dd3302f9c007a8a<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8476-ec63258.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8476-ec63258.dmg

**Comment by zzmp** [comment]
test this please

**Comment by zzmp** [comment]
On my machine this avoids the crash, but is not at all performant (lots of stutter). I am using AMD/Vive.

**Comment by zzmp** [comment]
I intend to check this install on an NVidia/Vive machine tomorrow. @imgntn if you could try it, lmk if you have the same slow experience with AMD/Vive.

**Comment by hifi-gustavo** [comment]
Build console output for commit: ec63258f8556fbf5fbb93769dbeb0cf90b81b5a6 available at: https://gist.github.com/ad709337264fa1fc6dd3302f9c007a8a<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8476-ec63258.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8476-ec63258.dmg

**Comment by imgntn** [comment]
this is aweosme -- no longer crashes on switch to steam VR!!! yay!  it was very performant on dev playa and some other places i went.

**Comment by zzmp** [comment]
QA

- Test on a Vive running on AMD and nVidia. Both must be OK for this to pass.
- Ensure that render performance stays the same.

**Comment by jherico** [comment]
We can't just acquire sync objects and never release them.  They're a resource that you have to release like any other.

**Comment by zzmp** [comment]
I can try replacing the syncs with ARB or EXT methods doing the same, and see if AMD can handle those.
Otherwise, I will start thinking about other ways to implement this driver, because it is unusable for AMD right now.

File: plugins/openvr/src/OpenVrDisplayPlugin.cpp [hunk 1]
```diff
@@ -111,12 +111,17 @@ class OpenVrSubmitThread : public QThread, public Dependency {
         _plugin.withNonPresentThreadLock([&] {
             while (!_queue.empty()) {
                 auto& front = _queue.front();
+
                 auto result = glClientWaitSync(front.fence, 0, 0);
-                if (GL_TIMEOUT_EXPIRED == result && GL_WAIT_FAILED == result) {
+
+                if (GL_TIMEOUT_EXPIRED == result || GL_WAIT_FAILED == result) {
                     break;
+                } else if (GL_CONDITION_SATISFIED == result || GL_CONDITION_SATISFIED == result) {
```
Comment (jherico) [inline plugins/openvr/src/OpenVrDisplayPlugin.cpp]: `GL_CONDITION_SATISFIED == result || GL_CONDITION_SATISFIED == result` is redundant.

**Comment by jherico** [comment]
fix the redundant conditional and it's fine.

**Comment by hifi-gustavo** [comment]
Build console output for commit: a67c5221e327950855a6cc70ade245d548bcba06 available at: https://gist.github.com/ad709337264fa1fc6dd3302f9c007a8a<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8476-a67c522.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8476-a67c522.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: dbb916e2d9f56b6ec158c528ce48221217798706 available at: https://gist.github.com/ad709337264fa1fc6dd3302f9c007a8a<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8476-dbb916e.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8476-dbb916e.dmg

**Comment by imgntn** [comment]
unfortunately this crashes the second time i switch to hmd mode (every time)

**Comment by zzmp** [comment]
This crashes do to a separate bug. I think this should be a QA Pass, as it fixes the single bug of a missing flush after the fence sync.

Incidentally, there is another bug which prevents AMD Vive users from switching displays more than once.

I think it is better to work some of the time than none of the time. Open to feedback.

**Comment by zzmp** [comment]
Also want to get this fix in so that it isn't forgotten about while we chase the next bug. Don't want someone to have to retrace it later.

**Comment by imgntn** [comment]
it does at least make it so you can use the HMD the first time

**Pull request merged**